lib/grape/endpoint.rb

```diff
@@ -232,6 +232,7 @@ def run(env)
 
       route_setting(:saved_validations).each do |validator|
         begin
+          validator.context = Hashie::Mash.new(request: request, cookies: cookies)
```

Comment:
I think the validator object belongs to the endpoint, which is never GCed, an endpoint is a class thing. So yes, you need to find a way to pass this along, maybe via ENV?